core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1007,7 +1010,8 @@ class ReplicaManager(val config: KafkaConfig,
       transactionalId,
       producerId,
       producerEpoch,
-      generalizedCallback
+      generalizedCallback,
+      ApiKeys.PRODUCE.latestVersion
```

Comment:
Hmm this is a bit interesting since this path is usually exercised by the group coordinator. We may want to do something else, or leave a comment here.

Reply:
I don't think that including the latest Produce Request Version will break any code flow. So should we just leave a comment denoting the same?

Reply:
What if we send a offset commit request that can't yet handle the error.
I think we may need a different data structure. Maybe something like enums to handle the various versions 
0 -- default, 1 -- new error, 2 -- add partitions rather than verifying.

Reply:
So for this particular path it would be 0 (default)
And for the new workflow we added, it would be new error (1)
Where can we use the 3rd use case --> that is add partitions rather than verifying?

Reply:
This will be done as a follow up and part of KIP-890 part 2
cc: @CalvinConfluent who was also thinking about this part